### PR TITLE
[REVIEW] Added factory methods for signon mode

### PR DESCRIPTION
**Pull request by lucasvanlierop**
This way no invalid modes can be created and notation is easier

@arnovr @remyhonig

### CODE REVIEW

**Comment by remyhonig** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ $client = new SingleSignOnClient(
 
 
 $handShakeUrl = $client->requestHandshakeUrl(
-    new SignOnMode(SignOnMode::USER),
+    new SignOnMode::asUser(),
```
Comment (arnovr) [inline README.md]: Static call which returns an instance, you could remove the "new"

File: src/DreamCommerce/SingleSignOn/SignOnMode.php [hunk 2]
```diff
@@ -10,7 +8,7 @@
 class SignOnMode
 {
     const SUPPORT_READ_ONLY = 0;
```
Comment (arnovr) [inline src/DreamCommerce/SingleSignOn/SignOnMode.php]: I still think the constants are weird
Shouldn't it be something like
READ_ONLY
READ_WRITE

File: README.md (hunk 1 above)
Comment (lucasvanlierop) [inline README.md]: Fixed, too bad this cannot be tested...

File: src/DreamCommerce/SingleSignOn/SignOnMode.php (hunk 2 above)
Comment (lucasvanlierop) [inline src/DreamCommerce/SingleSignOn/SignOnMode.php]: It was like that but then I realized that `USER` did not have the suffix either so I removed it from support also making it more clear (or so I thought) that one of the modes is specifically read only. Do you agree or still think this is weird?

Comment (arnovr) [inline src/DreamCommerce/SingleSignOn/SignOnMode.php]: Oke leave it as is

**Comment by arnovr** [comment]
:+1:

**Pull request merged**